backend/routes/api/export.js

```diff
@@ -416,6 +416,32 @@ router.get(
         // Get the Feature for this DateRange, if there is one
         const feature = getFeatureForDateRange(dateRange);
 
+        // Pre-calculate GateDetail
+        const isGateType = dateRange.dateType.name === "Operating (Gate)";
+        const hasGate = gateDetail?.hasGate === true;
+
+        let gateStartTime = "";
+        let gateEndTime = "";
+        let sameEveryYear = "";
```

Comment:
How about moving this logic to a dedicated function, similar to wh at we've done with `getFeatureForDateRange` and `getInReservationSystem`?

It could return an object with all of the values we need for this:

```suggestion
        // something like this
        const { gateStartTime, gateEndTime, sameEveryYear } = getGateDetails(season);
```